packages/assistify-threading/authorization.js

```diff
@@ -0,0 +1,23 @@
+import { Meteor } from 'meteor/meteor';
+import { RocketChat } from 'meteor/rocketchat:lib';
+import { addRoomAccessValidator, canAccessRoom } from 'meteor/rocketchat:authorization';
+import { Rooms } from 'meteor/rocketchat:models';
+
+Meteor.startup(() => {
+
+	// Add permissions for threading
+	const permissions = [
+		{ _id: 'start-thread', roles: ['admin', 'user', 'expert', 'guest'] },
+		{ _id: 'start-thread-other-user', roles: ['admin', 'user', 'expert', 'owner'] },
+	];
+
+	for (const permission of permissions) {
+		if (!RocketChat.models.Permissions.findOneById(permission._id)) {
+			RocketChat.models.Permissions.upsert(permission._id, { $set: permission });
+		}
+	}
+
+	addRoomAccessValidator(function(room, user) {
+		return room.prid && canAccessRoom(Rooms.findOne(room.prid), user);
```

Comment:
Ah, nice. I thought about this too. This makes it optional to invite users to the thread, which was a major flaw in the current implementation.

However, we got feedback that some users shall be able to join a thread while not being able to see the parent. Is this possible?